tap_pendo/__init__.py

```diff
@@ -151,14 +155,32 @@ def sync(config, state, catalog):
         singer.write_state(state)
     LOGGER.info("Finished sync")
 
+# To check that given number is numeric or not
+def isInteger(n):
+    try:
+        int(n)
+        return False
+    except Exception:
+        return True
+
 @utils.handle_top_exception(LOGGER)
 def main():
 
     # Parse command line arguments
     args = utils.parse_args(REQUIRED_CONFIG_KEYS)
 
+    # If app_ids is not in configure file then get all apps data
+    app_ids = args.config.get("app_ids", "expandAppIds(\"*\")").replace(" ", "") or "expandAppIds(\"*\")"
+    # If app_ids is given in configure file then check all app_ids are numeric or not
+    invalid_app_ids = []
+    if app_ids != "expandAppIds(\"*\")":
+        app_ids = app_ids.split(",")
+        invalid_app_ids = list(filter(isInteger, app_ids))
+
+    args.config["app_ids"] = app_ids
+
     if args.discover:
-        do_discover(args.config)
+        do_discover(args.config, invalid_app_ids)
```

Comment:
Raise exception for the invalid_app_ids here only. Either move the whole code in do_discover or put an exception here only.

Reply:
Raised exception for the invalid_app_ids inside the  main function